warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,138 @@
+                    break;
+                case DST_FIELD:
+                    whindexConfig.setDestField(entry.getValue());
+                    break;
+                case VALUES:
+                    whindexConfig.setValues(List.of(entry.getValue().split(",")));
+                    break;
+                default:
+                    String originalProperty = commonPrefix + groupID + "." + property;
+                    log.warn("Unexpected whindex property given:" + originalProperty + "=" + entry.getValue());
+            }
+        }
+
+        // add to fieldDefinitions and VFToWC
+        groupingsToConfigs.values().forEach((wc) -> {
+            valueFieldsToWhindexConfigs.put(wc.getValueField(), wc);
+        });
+
+    }
+
+    @Override
+    public Multimap<String,NormalizedContentInterface> getWhindexFields(Multimap<String,NormalizedContentInterface> eventMap) {
+
+        Multimap<String,NormalizedContentInterface> whindicesInEventMap = HashMultimap.create();
+
+        // Get all the wc that have both vf and sf in the eventMap
+        List<WhindexConfig> matchingConfigs = valueFieldsToWhindexConfigs.entries().stream()
+                        .filter(entry -> eventMap.containsKey(entry.getValue().getValueField()) && eventMap.containsKey(entry.getValue().getSourceField()))
+                        .map(Map.Entry::getValue).collect(Collectors.toList());
+
+        // Check that the eventMap entry has EITHER an EventField or IndexedField in common with the wc's VALUES
+        for (WhindexConfig curWhindexConfig : matchingConfigs) {
+            Collection<NormalizedContentInterface> relatedValueEventContents = eventMap.get(curWhindexConfig.getValueField());
+            // if any of the NCI's have either EF or IF that's GOOD!
+            boolean containsAnyMatchingValue = relatedValueEventContents.stream()
+                            .anyMatch(nci -> curWhindexConfig.getValues().contains(nci.getEventFieldValue())
+                                            || curWhindexConfig.getValues().contains(nci.getIndexedFieldValue()));
+
+            if (containsAnyMatchingValue) {
+                Collection<NormalizedContentInterface> relatedSourceEventContents = eventMap.get(curWhindexConfig.getSourceField());
+                List<NormalizedContentInterface> copies = new ArrayList<>();
+                for (NormalizedContentInterface content : relatedSourceEventContents) {
+                    NormalizedFieldAndValue copy = new NormalizedFieldAndValue(content);
+                    copies.add(copy);
+                }
+
+                // Create whindex fields
+                whindicesInEventMap.putAll(curWhindexConfig.getDestField(), copies);
+            }
+        }
+
+        return whindicesInEventMap;
+
+    }
+
+    @Override
+    public Multimap<String,WhindexConfig> getValueFieldsToWhindexConfigs() {
+        return valueFieldsToWhindexConfigs;
+    }
+
+    @Override
+    public boolean isWhindexField(String field) {
```

Comment:
I think it would be good for us to have a separate class level `Set<String>` variable that contains the dest fields that we populate during a call to `setup(Configuration config)`. It would make the implementation of this method simpler. 

Additionally, we should make this new set, the overloadedFields set, and the valueFieldsToWhindexConfigs map immutable/unmodifiable so that they cannot be modified after they've been retrieved via their getter methods. We should only support modification of them through calls to setup(), or through explicit setter methods.